modules/openapi-generator/src/main/resources/JavaSpring/enumOuterClass.mustache

```diff
@@ -46,6 +46,6 @@ public enum {{{datatypeWithEnum}}}{{^datatypeWithEnum}}{{{classname}}}{{/datatyp
         return b;
       }
     }
-    {{#isNullable}}return null;{{/isNullable}}{{^isNullable}}throw new IllegalArgumentException("Unexpected value '" + value + "'");{{/isNullable}}
+    {{#isNullable}}return null;{{/isNullable}}{{^isNullable}}{{#enumUnknownDefaultCase}}{{#allowableValues}}{{#enumVars}}{{#-last}}return {{{name}}};{{/-last}}{{/enumVars}}{{/allowableValues}}{{/enumUnknownDefaultCase}}{{^enumUnknownDefaultCase}}throw new IllegalArgumentException("Unexpected value '" + value + "'");{{/enumUnknownDefaultCase}}{{/isNullable}}
```

Comment:
What's the expected behaviour for when `isNullable` is true? Would you expect `null` or the default case? I would expect the default case if it was not explicitly set to null...

Reply:
I agree with this @welshm after investigating I feel it makes sense to return the default case. I was copying the functionality from the fix on #12344 for Java. Would it be right to make this function slightly differently then and return the DefaultCase when nullable is true?